File: src/main/java/com/shipout/dao/UserMapper.java
package com.shipout.dao;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import com.shipout.entity.User;
import org.apache.ibatis.annotations.Param;

import java.util.List;

public interface UserMapper extends BaseMapper<User> {

    void batchSaveUser(@Param("userList") List<User> userList);

    default void insertUser(@Param("user") User user) {
        if (user == null) {
            return;
        } else {
            insertUserNotNull(user);
        }
    }

    void insertUserNotNull(User user);
}

File: src/main/java/com/shipout/dao/UserMapper.kt
package com.shipout.dao

import com.baomidou.mybatisplus.core.mapper.*
import com.shipout.entity.*
import org.apache.ibatis.annotations.*

interface UserMapper : BaseMapper<User> {
    fun batchSaveUser(@Param("userList") userList: List<User>)

    fun insertUserNotNull(user: User)
}
